src/ErrorHandling/ErrorState.tsx

```diff
@@ -0,0 +1,86 @@
+import { EventEmitter } from "../utils/EventEmitter";
+import { Settings } from "../Settings/Settings";
+import { GetAllServers } from "../Server/AllServers";
+import { killWorkerScriptByPid } from "../Netscript/killWorkerScript";
+
+export type ErrorState = {
+  ErrorUpdate: EventEmitter<[ErrorRecord]>;
+  ActiveError: ErrorRecord | null;
+  Errors: ErrorRecord[];
+  ErrorPageOpen: boolean;
+  UnreadErrors: number;
+  PreventModals: boolean;
+};
+
+export type ErrorRecord = {
+  server: string;
+  errorType: string;
+  message: string;
+  scriptName: string;
+  pid: number;
+  occurrences: number;
+  time: Date;
+  unread: boolean;
+  force?: boolean;
+};
+
+export const ErrorState: ErrorState = {
+  ErrorUpdate: new EventEmitter<[ErrorRecord]>(),
+  ActiveError: null as ErrorRecord | null,
+  Errors: [],
+  ErrorPageOpen: false,
+  UnreadErrors: 0,
+  PreventModals: Settings.SuppressErrorModals,
+};
```

Comment:
```suggestion
export const ErrorState = {
  ErrorUpdate: new EventEmitter<[ErrorRecord]>(),
  ActiveError: null as ErrorRecord | null,
  Errors: [] as ErrorRecord[],
  ErrorPageOpen: false,
  UnreadErrors: 0,
  PreventModals: Settings.SuppressErrorModals,
};
```
With this change, you don't need to specify the type above.